src/gaxios.ts

```diff
@@ -416,4 +431,30 @@ export class Gaxios {
       return response.blob();
     }
   }
+
+  /**
+   * Creates an async generator that yields the pieces of a multipart/related request body.
+   *
+   * @param {GaxioMultipartOptions[]} multipartOptions the pieces to turn into a multipart/related body.
+   * @param {string} boundary the boundary string to be placed between each part.
+   */
+  private async *getMultipartRequest(
```

Comment:
Is this based on a specific spec, or is this just what we need in Storage?

Reply:
This is based on https://www.ietf.org/rfc/rfc2387.txt with a slight caveat. We are not currently supporting multipart/related within a part (i.e. recursive support).

Reply:
sg. Do you mind adding a note in the comments?

Reply:
Done